cocos/3d/framework/skinning-model-component.js

```diff
@@ -80,40 +93,31 @@ export default class SkinningModelComponent extends ModelComponent {
      * !#ch 骨骼节点
      * @type {Skeleton}
      */
-    @property({
-        type: Skeleton
-    })
+    @property(Skeleton)
     get skeleton() {
         return this._skeleton;
     }
 
     set skeleton(val) {
         this._skeleton = val;
-
-        if (this._skeleton) {
-            this._skeletonInstance = new SkeletonInstance(this._skeleton);
-            this._reInitJointsData();
-        } else {
-            this._skeletonInstance = null;
-        }
-    }
-
-    /**
-     * !#en The bone nodes
-     *
-     * !#ch 实例化骨骼节点
-     * @type {SkeletonInstance}
-     */
-    get skeletonInstance() {
-        return this._skeletonInstance;
     }
 
     onLoad() {
+        this._reInitJointsData();
         this._updateModels();
         this._updateCastShadow();
         this._updateReceiveShadow();
 
-        //SkinningModelComponent.system.add(this);
+        /** @type {import("../assets/material").default} */
+        let mtl = new cc.Material();
+        mtl.effect = cc.game._builtins['builtin-effect-unlit'];
+        mtl.setProperty("color", new cc.vmath.color4(0, 0, 0, 1));
+        this.material = mtl;
+    }
+
+    onEnable() {
+        this._resetTarget();
```

Comment:
should we recreate `skinningTarget` every time when enabled? could we create it when it is loaded?

Reply:
I've done it.